src/lib.rs

```diff
@@ -121,10 +148,17 @@ impl UefiBoot {
     /// Creates an UEFI-bootable FAT partition with the kernel.
     fn create_fat_partition(&self) -> anyhow::Result<NamedTempFile> {
         let bootloader_path = Path::new(env!("UEFI_BOOTLOADER_PATH"));
-
+        let has_rd_path = self.ramdisk.is_some();
+        let binding = self.ramdisk.as_deref();
+        let ramdisk_path = binding.unwrap_or(Path::new("no-such-file"));
+        let kernel_path = self.kernel.as_path();
         let mut files = BTreeMap::new();
         files.insert("efi/boot/bootx64.efi", bootloader_path);
-        files.insert(KERNEL_FILE_NAME, self.kernel.as_path());
+        files.insert(KERNEL_FILE_NAME, kernel_path);
+
+        if has_rd_path {
+            files.insert(RAMDISK_FILE_NAME, ramdisk_path);
+        }
```

Comment:
Same as above (use `if let` and avoid the additional bindings in lines 151-154).